File: sdks/dotnet/src/Resources/WebhooksResource.cs
using System.Collections.Generic;
using System.Threading.Tasks;
using Puxbay.SDK.Models;

namespace Puxbay.SDK.Resources
{
    public class WebhooksResource
    {
        private readonly PuxbayClient _client;
        public WebhooksResource(PuxbayClient client) { _client = client; }

        public async Task<PaginatedResponse<Webhook>> ListAsync(int page = 1)
        {
            return await _client.GetAsync<PaginatedResponse<Webhook>>($"webhooks/?page={page}");
        }

        public async Task<Webhook> GetAsync(string webhookId)
        {
            return await _client.GetAsync<Webhook>($"webhooks/{webhookId}/");
        }

        public async Task<Webhook> CreateAsync(Webhook webhook)
        {
            return await _client.PostAsync<Webhook>("webhooks/", webhook);
        }

        public async Task<Webhook> UpdateAsync(string webhookId, Webhook webhook)
        {
            return await _client.PatchAsync<Webhook>($"webhooks/{webhookId}/", webhook);
        }

        public async Task DeleteAsync(string webhookId)
        {
            await _client.DeleteAsync($"webhooks/{webhookId}/");
        }
        
        public async Task<Dictionary<string, object>> ListEventsAsync(string webhookId, int page = 1)
        {
            return await _client.GetAsync<Dictionary<string, object>>($"webhook-logs/?webhook={webhookId}&page={page}");
        }
    }
}

File: sdks/php/src/Resources/WebhooksResource.php
<?php

namespace Puxbay\Resources;

use Puxbay\Models\Webhook;
use Puxbay\Models\PaginatedResponse;

class WebhooksResource extends BaseResource
{
    public function list(int $page = 1): PaginatedResponse
    {
        $data = $this->get("webhooks/?page={$page}");
        return PaginatedResponse::fromArray($data, fn($item) => Webhook::fromArray($item));
    }

    public function getWebhook(string $webhookId): Webhook
    {
        $data = $this->get("webhooks/{$webhookId}/");
        return Webhook::fromArray($data);
    }

    public function create(Webhook $webhook): Webhook
    {
        $data = $this->post('webhooks/', $webhook->toArray());
        return Webhook::fromArray($data);
    }

    public function update(string $webhookId, Webhook $webhook): Webhook
    {
        $data = $this->patch("webhooks/{$webhookId}/", $webhook->toArray());
        return Webhook::fromArray($data);
    }

    public function deleteWebhook(string $webhookId): void
    {
        $this->delete("webhooks/{$webhookId}/");
    }

    public function listEvents(string $webhookId, int $page = 1): array
    {
        // Events are usually just raw data logs, keep as array or make Event DTO
        return $this->get("webhook-logs/?webhook={$webhookId}&page={$page}");
    }
}
